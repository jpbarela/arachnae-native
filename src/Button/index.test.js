// @flow
import * as React from "react";
import { render } from "@testing-library/react-native";
import { Button } from "./index";

describe("Button", () => {
  it("renders with no style", () => {
    const container = render(<Button title="Press Me" />);
    expect(container).toMatchSnapshot();
  });

  it("renders with link style", () => {
    const container = render(<Button title="Click Me" style="link" />);
    expect(container).toMatchSnapshot();
  });

  it("renders with button style", () => {
    const container = render(<Button title="Press Me" style="button" />);
    expect(container).toMatchSnapshot();
  });
});
