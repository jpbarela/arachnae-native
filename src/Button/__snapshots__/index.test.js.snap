// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders with button style 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {},
      Object {},
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": undefined,
          "fontFamily": "arial",
        },
        null,
        Object {},
      ]
    }
  >
    Press Me
  </Text>
</View>
`;

exports[`Button renders with link style 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {},
      Object {},
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": undefined,
          "fontFamily": "arial",
        },
        null,
        Object {},
      ]
    }
  >
    Click Me
  </Text>
</View>
`;

exports[`Button renders with no style 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {},
      Object {},
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": undefined,
          "fontFamily": "arial",
        },
        null,
        Object {},
      ]
    }
  >
    Press Me
  </Text>
</View>
`;
